src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,324 @@
+        validateAndParseAlgorithmMap(algorithmMap);
+        this.fieldMap = fieldMap;
+        this.environment = environment;
+        this.clusterService = clusterService;
+        this.indicesService = indicesService;
+        this.analysisRegistry = analysisRegistry;
+    }
+
+    public String getType() {
+        return TYPE;
+    }
+
+    @SuppressWarnings("unchecked")
+    private void validateAndParseAlgorithmMap(Map<String, Object> algorithmMap) {
+        if (algorithmMap.size() != 1) {
+            throw new IllegalArgumentException(
+                "Unable to create the processor as [" + ALGORITHM_FIELD + "] must contain and only contain 1 algorithm"
+            );
+        }
+
+        for (Map.Entry<String, Object> algorithmEntry : algorithmMap.entrySet()) {
+            String algorithmKey = algorithmEntry.getKey();
+            Object algorithmValue = algorithmEntry.getValue();
+            if (!supportedChunkers.contains(algorithmKey)) {
+                throw new IllegalArgumentException(
+                    "Unable to create the processor as chunker algorithm ["
+                        + algorithmKey
+                        + "] is not supported. Supported chunkers types are ["
+                        + FIXED_LENGTH_ALGORITHM
+                        + ", "
+                        + DELIMITER_ALGORITHM
+                        + "]"
+                );
+            }
+            if (!(algorithmValue instanceof Map)) {
+                throw new IllegalArgumentException(
+                    "Unable to create the processor as [" + ALGORITHM_FIELD + "] cannot be cast to [" + Map.class.getName() + "]"
+                );
+            }
+            FieldChunker chunker = ChunkerFactory.create(algorithmKey, analysisRegistry);
+            chunker.validateParameters((Map<String, Object>) algorithmValue);
+            this.chunkerType = algorithmKey;
+            this.chunkerParameters = (Map<String, Object>) algorithmValue;
+            chunker.validateParameters(chunkerParameters);
+            if (((Map<String, Object>) algorithmValue).containsKey(MAX_CHUNK_LIMIT_FIELD)) {
+                this.max_chunk_limit = ((Number) ((Map<String, Object>) algorithmValue).get(MAX_CHUNK_LIMIT_FIELD)).intValue();
+            }
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private boolean isListString(Object value) {
+        // an empty list is also List<String>
+        if (!(value instanceof List)) {
+            return false;
+        }
+        for (Object element : (List<Object>) value) {
+            if (!(element instanceof String)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private List<String> chunkString(String content) {
+        FieldChunker chunker = ChunkerFactory.create(chunkerType, analysisRegistry);
+        List<String> result = chunker.chunk(content, chunkerParameters);
+        current_chunk_count += result.size();
```

Comment:
We need to reset it to 0 when a new document input.

Reply:
Thanks! Already changed.